### DISCUSSION TITLE
Estimates on when features will arrive to rockchip on kernel 6.5

### DISCUSSION THREAD

**Discussion by jgresham** [Joshua-Riek/ubuntu-rockchip#372] [General]
Just curious if anyone has insight or educated guesses on when we will see features like HDMI out on rockchip's using linux kernel 6.5? 
What other features still need to be worked on for kernel 6.5?

**Comment by Joshua-Riek** [comment]
More features most likely will not be coming to the 6.5 kernel unless someone backports them. 

Since many hardware features are not yet supported, it makes sense to use the absolute latest kernel available until we have most of the hardware features working such as HDMI. 

I'll probably be coming out with some 6.6-rc2 kernel builds this weekend, 6.5 will be left in the dust, so to speak.

**Reply by jgresham** [reply]
thanks for info! good to hear.

I wish I knew more about kernel development so I could try to help with hdmi out and more. Curious, have you been able to merge features into the linux mainline?

**Reply by Joshua-Riek** [reply]
I have not merged anything into the mainline kernel yet. I'm still learning but confident when it comes to device trees and bringing up new hardware. I need to look into driver development at some point so I can do more.

**Reply by jgresham** [reply]
It looks like `6.9-rc1` includes a RK3588 PHY driver for HDMI/eDP https://github.com/torvalds/linux/commit/23ed255e74317b10b41b396d4af5b79b47b28c22

Does this mean HDMI out on mainline for RK3588?

**Comment by jgresham** [comment]
It looks like there has been some HDMI out work for RK3588 on gitlab https://gitlab.collabora.com/hardware-enablement/rockchip-3588/linux/-/commits/rk3588. Also looks like NanoPC t6 official images were updated to 6.1kernel recently https://wiki.friendlyelec.com/wiki/index.php/NanoPC-T6#Install_OS.

Any chance this means that HDMI out for RK3588 on mainline linux is near?

**Reply by Joshua-Riek** [reply]
I have some experimental HDMI images for the Radxa Rock 5B and Orange Pi 5 boards. But it is very buggy, and I've noticed a lot of system crashes related to the HDMI driver. 

https://github.com/Joshua-Riek/ubuntu-rockchip/actions/workflows/build-mainline.yml

**Reply by Joshua-Riek** [reply]
As for the 6.1 kernel, there is still no official release, I have been checking the SDK server daily. I think it is a poor choice on FriendlyELEC's part to use an experimental development 6.1 branch.
